### ISSUE TITLE
Variables within closures return wrong values

### ISSUE THREAD

**Issue by jakegut** [risor-io/risor#82]
Given:

```go
func foo(a, b){
    return func(c){
        return [a, b, c]
    }
}

foo("hello", "world")("risor")
```

I expected `["hello", "world", "risor"]`, instead I get: `["world","hello","risor"]`. Reproduced using the code editor on the home page.

I'd be curious to look into this myself and see if I can PR a fix.

**Comment by myzie** [comment]
Great find and great fix 🏅 

Fixed by https://github.com/risor-io/risor/pull/83

**Comment by myzie** [comment]
This is now available in the v0.13.0 release, and on the editor on https://risor.io.
